src/facet/inspection/_shap_projection.py

```diff
@@ -221,7 +264,36 @@ def _fit(self, shap_calculator: ShapInteractionValuesCalculator) -> None:
         # Redundancy: red[i, j]
         #
 
-        red_ij = ass_ij - syn_ij
+        # cov(p[i], p[j])
+        # covariance matrix of shap vectors
+        # shape: (n_outputs, n_features, n_features)
+        cov_p_i_p_j = cov(p_i, weight)
+
+        # nominator
+        # shape: (n_outputs, n_features, n_features)
+        red_ij_nominator = (
+            cov_p_i_p_j - cov_p_i_p_ij_over_var_p_ij * transpose(cov_p_i_p_ij)
+        ) ** 2
+
+        # denominator for p_i
+        # shape: (n_outputs, n_features, n_features)
+        red_ij_denominator_i = var_p_i - cov_p_i_p_ij_over_var_p_ij * cov_p_i_p_ij
+        red_ij_denominator = red_ij_denominator_i * transpose(red_ij_denominator_i)
+
+        # red[i, j]
+        # this converges towards 0 as the denominator converges towards 0
+        # shape: (n_outputs, n_features, n_features)
+
+        red_ij = np.zeros(matrix_shape)
+        np.divide(
+            red_ij_nominator,
+            red_ij_denominator,
+            out=red_ij,
+            where=red_ij_denominator > 0.0,
```

Comment:
See above

Reply:
See comments above, all "ingredients" of the division will have a similar order of magnitude.